components/tidb_query_datatype/src/codec/data_type/vector.rs

```diff
@@ -37,6 +37,21 @@ impl VectorValue {
         }
     }
 
+    /// Creates a `VectorValue` of length `len` with the given value `scalar`.
+    #[inline]
+    pub fn from_scalar(scalar: &ScalarValue, len: usize) -> Self {
+        macro_rules! expand_convertion {
+            ($val:tt, $( $tp:tt ),* ) => {
+                match &$val {
+                    $(
+                        &ScalarValue::$tp(val) => { VectorValue::$tp(ChunkedVec::from_vec(vec![val.clone(); len])) }
```

Comment:
This is not efficient. If we want to create a chunked vec from repeated val, use `append` instead of constructing from `vec!`.

Reply:
you mean push?

Reply:
Yes!

Reply:
okay, let me try to refactor it a little bit.